pydantic/validators.py

```diff
@@ -99,8 +100,13 @@ def bool_validator(v: Any) -> bool:
     if isinstance(v, bytes):
         v = v.decode()
     if isinstance(v, str):
-        return v.upper() in BOOL_STRINGS
-    return bool(v)
+        v = v.lower()
+    with change_exception(errors.BoolError, TypeError):
```

Comment:
can we have a test for this failure.

Reply:
Done (the test is called `test_bool_unhashable_fails`); let me know if I've misinterpreted your intention.